Makefile

```diff
@@ -298,7 +298,7 @@ ifneq ($(wildcard $(XSMM_DIR)/lib/libxsmm.*),)
   ifeq (,$(MKL)$(MKLROOT))
     BLAS_LIB = -lblas
   else
-    BLAS_LIB = $(if $(MKLROOT),-L$(MKLROOT)/lib/intel64 -Wl,-rpath,$(MKLROOT)/lib/intel64) -Wl,--no-as-needed -lmkl_intel_lp64 -lmkl_sequential -lmkl_core -lpthread -lm -ldl
+    BLAS_LIB = $(if $(MKLROOT),-L$(MKLROOT)/lib/intel64 -Wl,-rpath,$(MKLROOT)/lib/intel64 -Wl),--no-as-needed -lmkl_intel_lp64 -lmkl_sequential -lmkl_core -lpthread -lm -ldl
```

Comment:
I think the issue is that the commas are controls for the `$(if`. Quotes or a variable containing `-L$(MKLROOT)/lib/intel64 -Wl,-rpath,$(MKLROOT)/lib/intel64` should fix it.

Reply:
Quotes won't do:
```
$ make print-BLAS_LIB MKLROOT=/foo
[ variable name]: BLAS_LIB
[        origin]: file
[         value]: $(if $(MKLROOT),"-L$(MKLROOT)/lib/intel64 -Wl,-rpath,$(MKLROOT)/lib/intel64 -Wl"),--no-as-needed -lmkl_intel_lp64 -lmkl_sequential -lmkl_core -lpthread -lm -ldl
[expanded value]: "-L/foo/lib/intel64 -Wl,--no-as-needed -lmkl_intel_lp64 -lmkl_sequential -lmkl_core -lpthread -lm -ldl
```

This is missing the rpath part. Just put it in a separate variable.

Reply:
Thanks, I hadn't tested it yet. I'll fix it up in the morning.